### ISSUE TITLE
soup migrate turns axolotl `rl: orpo`, `ipo` and `simpo` into `task: sft` with no warning; LLaMA-Factory `pref_loss: ipo` becomes plain DPO

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#1214]
## What happens

`soup migrate --from axolotl` maps the source's `rl:` value through a four-entry table and falls back to `sft` for everything else. Soup has `orpo`, `ipo` and `simpo` tasks, but those three axolotl values all migrate to `task: sft`, and no warning mentions `rl`.

Minimal repro, on `main` (`99d553f5`), CPU only:

```python
from pathlib import Path
from soup_cli.config.loader import load_config_from_string
from soup_cli.migrate.axolotl import migrate_axolotl
from soup_cli.migrate.common import config_to_yaml

p = Path("axo.yaml"); p.write_text("base_model: m\nrl: orpo\n", encoding="utf-8")
r = migrate_axolotl(p)
print(r["task"], r["_warnings"], load_config_from_string(config_to_yaml(r)).task)
# sft ['No datasets specified. Using placeholder path.'] sft
```

The only warning is about the missing `datasets:` block. The same code over every value of axolotl's own `RLType` enum (`dpo, gdpo, grpo, ipo, orpo, kto, simpo, ebft`):

```
axolotl rl: dpo    migrated task=dpo   -> LOADS task=dpo
axolotl rl: kto    migrated task=kto   -> LOADS task=kto
axolotl rl: grpo   migrated task=grpo  -> LOADS task=grpo
axolotl rl: gdpo   migrated task=grpo  -> LOADS task=grpo
axolotl rl: orpo   migrated task=sft   -> LOADS task=sft     (no warning about rl)
axolotl rl: ipo    migrated task=sft   -> LOADS task=sft     (no warning about rl)
axolotl rl: simpo  migrated task=sft   -> LOADS task=sft     (no warning about rl)
axolotl rl: ebft   migrated task=sft   -> LOADS task=sft     (no warning about rl; no Soup equivalent)
```

The CLI prints nothing about the objective either:

```
$ soup migrate --from axolotl axo_rl_orpo.yaml --dry-run      (exit 0; panel borders stripped)
Migration Warnings
  ! No datasets specified. Using placeholder path.
Generated soup.yaml (from axolotl)
  base: m
  task: sft
  data:
    train: ./data/train.jsonl
    format: auto
  training: {}
  output: ./output
Dry run -- no file written.
```

With the dataset types axolotl actually uses for these objectives, the failure you see names a different field. The preference tasks have no entry in the task-to-format table, so the dataset type passes through unchanged and the schema refuses it:

```
axolotl rl: orpo, datasets type chat_template.argilla -> migrated task=sft -> REFUSED at load:
    data -> format: Input should be 'alpaca', 'sharegpt', 'chatml', 'dpo', ...
axolotl rl: ipo,  datasets type chatml.intel          -> migrated task=sft -> REFUSED at load: (same)
axolotl rl: simpo, datasets type chat_template        -> migrated task=sft -> LOADS task=sft
```

If you repair the format the error names (`format: dpo`), the file loads as `task: sft` and a real `soup train` (tiny random Llama, CPU) then fails in setup with an error that does not mention the task:

```
after format: dpo -> LOADS task: sft
$ soup train --config axo_orpo_repaired.yaml --yes    (exit 1)
    Task:    sft
ValueError("Column to remove ['messages'] not in the dataset. Current columns in the dataset: ['prompt', 'chosen', 'rejected']")
```

### The other two migrators have the same gap

LLaMA-Factory runs every preference loss under `stage: dpo` and selects it with `pref_loss` (`sigmoid, hinge, ipo, kto_pair, orpo, simpo`). Only three of those six are mapped. `ipo` loads as `task: dpo` even though Soup has an `ipo` task, and nothing warns about it:

```
LF stage dpo, pref_loss: sigmoid  -> LOADS task=dpo
LF stage dpo, pref_loss: orpo     -> LOADS task=orpo
LF stage dpo, pref_loss: simpo    -> LOADS task=simpo
LF stage dpo, pref_loss: ipo      -> LOADS task=dpo      warnings: none about the loss
LF stage dpo, pref_loss: hinge    -> LOADS task=dpo      (no Soup equivalent, no warning)
LF stage dpo, pref_loss: kto_pair -> LOADS task=dpo      (no Soup equivalent, no warning)
```

The Unsloth notebook migrator infers the task from the TRL trainer class, and any class not in its table stays `sft`:

```
unsloth DPOTrainer(...)       -> task=dpo      unsloth ORPOTrainer(...)      -> task=orpo
unsloth CPOTrainer(...)       -> task=sft      unsloth RewardTrainer(...)    -> task=sft
unsloth BCOTrainer(...)       -> task=sft      unsloth OnlineDPOTrainer(...) -> task=sft
```

The same class table also drives hyperparameter extraction, and it does not list TRL's `SFTConfig`. A notebook that passes `args=SFTConfig(...)` loses its batch size, epochs, learning rate and output directory with no warning. The same kwargs under `TrainingArguments(...)` survive:

```
args=TrainingArguments  migrated training={'batch_size': 2, 'epochs': 3, 'lr': 0.0002, 'quantization': '4bit'} output=outputs
args=SFTConfig          migrated training={'quantization': '4bit'} output=./output
```

## Why

- `src/soup_cli/migrate/axolotl.py:12-17`: `_RL_MAP` has four keys (`dpo, kto, grpo, gdpo`). Line 62 is `_RL_MAP.get(rl_type, "sft")`, which silently turns every other value into `sft`. `_TASK_FORMAT_MAP` (`:33-38`) has no preference-task entry, so line 76 (`_FORMAT_MAP.get(ds_type, ds_type)`) passes an axolotl-only dataset type straight into `data.format`.
- `src/soup_cli/migrate/llamafactory.py:21-25`: `_PREF_LOSS_MAP` maps `sigmoid`, `orpo` and `simpo`. For any other `pref_loss`, lines 65-67 keep the stage's task (`dpo`).
- `src/soup_cli/migrate/unsloth.py:12-28`: `_TRAINER_MAP` and `_CONFIG_MAP` list six trainer classes and five config classes. `task = "sft"` (`:84`) is kept for anything else, and hyperparameters are read only when the call is `TrainingArguments` or a `_CONFIG_MAP` class (`:132`).

In all three migrators, a source value the table does not list takes a default with no warning, and nothing downstream can tell that a default was taken.

## Fix path

Map every value the source tools document, and refuse the ones Soup cannot express instead of defaulting them. For axolotl: `orpo -> orpo`, `ipo -> ipo`, `simpo -> simpo`, each with `data.format: dpo` added to `_TASK_FORMAT_MAP`. `ebft` gets a refusal that names it. For LLaMA-Factory: `pref_loss: ipo -> ipo`, and `hinge` / `kto_pair` are refused or warned by name, because Soup has no equivalent. For Unsloth: `RewardTrainer -> reward_model`, `BCOTrainer -> bco`, `OnlineDPOTrainer -> online_dpo` (with a warning when its judge or reward model cannot be carried over). `CPOTrainer` maps to `simpo` only when its config sets `loss_type="simpo"`; otherwise it is refused. Add `SFTConfig` (and the other `*Config` classes) to the hyperparameter table.

Replace each `.get(value, "sft")` with a lookup that raises `ValueError` naming the unknown value and listing the supported ones, so `soup migrate` exits 1 with that message. A warning plus `auto` would also do for an unknown dataset `type`, instead of passing it through to a field the schema will refuse. While there, note that the preference objectives' own parameters (axolotl `rl_beta`, `orpo_alpha`, `simpo_gamma`, `cpo_alpha`) are not read either. That was not measured here and may be worth a follow-up.

## Acceptance

- [ ] A table-driven test over every axolotl `RLType` value (`dpo, gdpo, grpo, ipo, orpo, kto, simpo, ebft`), every LLaMA-Factory `pref_loss` (`sigmoid, hinge, ipo, kto_pair, orpo, simpo`) and the TRL trainer classes above (`CPOTrainer, RewardTrainer, BCOTrainer, OnlineDPOTrainer`, plus the six already mapped). Each value either produces the named Soup task, asserted on the loaded config (`load_config_from_string(config_to_yaml(result)).task`) rather than the migration dict, or is refused with the source value in the message. `rl: orpo` / `ipo` / `simpo` and `pref_loss: ipo` fail on current code.
- [ ] `rl: orpo` with `type: chat_template.argilla` produces a config that loads as `task: orpo`, `data.format: dpo`.
- [ ] A notebook passing `args=SFTConfig(per_device_train_batch_size=2, learning_rate=2e-4, num_train_epochs=3)` migrates those three values. This fails on current code.
- [ ] A CLI test: `soup migrate --from axolotl` on `rl: ebft` exits non-zero, and the output, asserted through the ANSI-strip + whitespace-collapse helper, names `ebft`.
